system-addon/lib/SectionsManager.jsm

```diff
@@ -162,6 +170,13 @@ const SectionsManager = {
               description: card.description,
               previewImageURL: card.image
             });
+            this.emit(this.DISPATCH_TO_MAIN, {
```

Comment:
I'm not sure what's the expected / allowed `emit`s. @k88hudson ? Wherever we have this behavior, it will still want a comment as previously requested.

Reply:
I'm confused as to why this needs to be emitted/added to PlacesFeed, why can't you just call ` PlacesUtils.history.insert` directly? Also is this the correct behaviour if the item is already in the history?

Reply:
@k88hudson my thinking was that doing the insert in PlacesFeed would allow us to have more control over the action in the future (like checking if the website has already been visited).
Regarding your second question this will now only happen for topstories, highlights are skipped over. I don't see any problem in having the extra visit.

Reply:
Personally that seems like premature abstraction to me, but if you are going to do it that way I'd probably go with a little more encapsulation and have a specific ADD_URL event rather than a generic DISPATCH_TO_MAIN

Reply:
I'll revert to directly calling `PlacesUtils.history.insert`.